jsapp/js/dataInterface.es6

```diff
@@ -491,10 +491,18 @@ export var dataInterface;
       if (fields.length)
         f = `&fields=${JSON.stringify(fields)}`;
 
-      return $ajax({
-        url: `${ROOT_URL}/api/v2/assets/${uid}/data/?${query}${s}${f}${filter}`,
-        method: 'GET'
-      });
+      if (fields.length > 3) {
+        const query_no_limit = `&start=${page}`;
+        return $ajax({
+          url: `${ROOT_URL}/api/v2/assets/${uid}/data/?${query_no_limit}${s}${f}${filter}`,
+          method: 'GET'
+        });
+      } else {
+        return $ajax({
+          url: `${ROOT_URL}/api/v2/assets/${uid}/data/?${query}${s}${f}${filter}`,
+          method: 'GET'
+        });
+      }
```

Comment:
If you pass `limit=0` in the url, it will return all submissions. I think that instead of putting this logic in here, it would be better to do it in `map.es6` where you call `dataInterface.getSubmissions(this.props.asset.uid, 5000, 0, sort, fq)`. Just add the `fields.length > 3` logic there and pass either `5000` or `0` as limit :)

Reply:
In this way the map-related change will be in the map file and it will be easier to understand this code in the future :)

Reply:
Already did in https://github.com/kobotoolbox/kpi/pull/2560 ! I tried to put `limit = 0` but it throws me an error saying `{"limit":"must be a positive integer"}`